File: 2024/programacao_avancada/2024.2/30-09-2024/1076.cpp
#include <bits/stdc++.h>

using namespace std;

vector<vector<int>> adj;
vector<bool> vis;

int cont = 0;

void dfs(int v){
    vis[v] = true;
    for (int x: adj[v]){
        if (!vis[x]){
            dfs(x);
            cont += 2;
        }
    }
}

void solve(){
    int n, v, a;
    cin >> n >> v >> a;
    adj.assign(v, vector<int>());
    vis.assign(v, false);
    for (int i = 0; i < a; i++){
        int x, y;
        cin >> x >> y;
        adj[x].push_back(y);
        adj[y].push_back(x);
    }
    dfs(n);
    cout << cont << "\n";
    cont = 0;
}

int main(){
    int t;
    cin >> t;
    while (t--){
        solve();
    }
    return 0;
}

File: 2025/intermediario I/2025.2/12-11-2025/1076.py
import sys
sys.setrecursionlimit(10**6)

def dfs(u, grafo, vis):
    vis[u] = True
    cont = 0
    for v in grafo[u]:
        if not vis[v]:
            cont += dfs(v, grafo, vis) + 2
    return cont

def solve():
    n = int(input())
    v, a = map(int, input().split())
    grafo = [[] for _ in range(v)]
    vis = [False] * v
    for _ in range(a):
        x, y = map(int, input().split())
        grafo[x].append(y)
        grafo[y].append(x)

    cont = dfs(n, grafo, vis)
    print(cont)

t = int(input())
for _ in range(t):
    solve()
